common/src/main/java/dev/mayaqq/estrogen/registry/blocks/DreamBlock.java

```diff
@@ -50,9 +52,26 @@ public boolean canBeReplaced(BlockState state, Fluid fluid) {
         return Shapes.block();
     }
 
+    /**
+     * Checks for if the player is colliding with a dream block.
+     */
     public static boolean isInDreamBlock(Player player) {
-        BlockPos pos = player.blockPosition();
-        return player.level().getBlockState(pos).getBlock() instanceof DreamBlock || player.level().getBlockState(pos.above()).getBlock() instanceof DreamBlock;
+        AABB playerAABB = player.getBoundingBox();
+        BlockPos minPos = BlockPos.containing(playerAABB.minX, playerAABB.minY, playerAABB.minZ);
+        BlockPos maxPos = new BlockPos(
+                Mth.ceil(playerAABB.maxX) - 1,
+                Mth.ceil(playerAABB.maxY) - 1,
+                Mth.ceil(playerAABB.maxZ) - 1
+        );
+        return BlockPos.betweenClosedStream(minPos, maxPos).anyMatch(
+                pos -> player.level().getBlockState(pos).getBlock() instanceof DreamBlock
+        );
+
+        // can't use betweenClosedStream because it also sometimes includes blocks that the player
```

Comment:
oops this is supposed to say "can't use betweenClosedStream(AABB aabb) etc. etc."